`./hack/.ci/prepare_release` is broken

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
"/priority" identifiers: 1|2|3|4|5 (ordered from greatest to least)
-->
/kind bug
/priority 2

**What happened**:
`./hack/.ci/prepare_release` that is used by the extension controllers currently fails with:

```
Downloading go 1.16.3
Executing make on go 1.16.3
> Installing requirements
golangci/golangci-lint info checking GitHub for tag 'v1.27.0'
golangci/golangci-lint info found version: 1.27.0 for v1.27.0/linux/amd64
golangci/golangci-lint info installed /tmp/tmp.tQZV8qPmu2/bin/golangci-lint
Downloading https://get.helm.sh/helm-v2.17.0-linux-amd64.tar.gz
Preparing to install helm and tiller into /usr/local/bin
helm installed into /usr/local/bin/helm
tiller installed into /usr/local/bin/tiller
Run 'helm init' to configure helm.
> Generate
Successfully generated controller registration at ../../example/controller-registration.yaml
Generating deepcopy funcs
Generating defaulters
Generating conversions
Generating deepcopy funcs
Generating defaulters
Generating conversions
I0512 14:40:46.706794   13998 main.go:127] parsing go packages in directory .
I0512 14:40:47.537866   13998 main.go:229] using package=github.com/gardener/gardener-extension-provider-azure/pkg/apis/azure/v1alpha1
I0512 14:40:47.545133   13998 main.go:165] written to ../../../../hack/api-reference/api.md
I0512 14:40:47.560662   14018 main.go:127] parsing go packages in directory .
I0512 14:40:48.375380   14018 main.go:229] using package=github.com/gardener/gardener-extension-provider-azure/pkg/apis/config/v1alpha1
I0512 14:40:48.822498   14018 main.go:165] written to ../../../../hack/api-reference/config.md
../../../../charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
../../../../charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
../tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
prog.go:12:2: cannot find package "github.com/golang/mock/mockgen/model" in any of:
	/usr/lib/go/src/github.com/golang/mock/mockgen/model (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/github.com/golang/mock/mockgen/model (from $GOPATH)
2021/05/12 14:40:51 Loading input failed: exit status 1
pkg/mock/factory/doc.go:14: running "mockgen": exit status 1
../../../../charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
../../../../charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
../tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/charts/images.go:17:8: cannot find package "embed" in any of:
	/tmp/tmp.tQZV8qPmu2/src/github.com/gardener/gardener-extension-provider-azure/vendor/embed (vendor tree)
	/usr/lib/go/src/embed (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/embed (from $GOPATH)
prog.go:12:2: cannot find package "github.com/golang/mock/mockgen/model" in any of:
	/usr/lib/go/src/github.com/golang/mock/mockgen/model (from $GOROOT)
	/tmp/tmp.tQZV8qPmu2/src/github.com/golang/mock/mockgen/model (from $GOPATH)
2021/05/12 14:40:52 Loading input failed: exit status 1
pkg/mock/vmss/doc.go:14: running "mockgen": exit status 1
make: *** [Makefile:124: generate] Error 1
```

Refs to failed releases:
- https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-extension-provider-azure-master/jobs/master-release-job/builds/2
- https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-extension-provider-azure-master/jobs/master-release-job/builds/2.1

The error is also locally reproducible.

Apparently there is a regression in the base image or in the pkgs that the base image or the script fetches as we haven't modify the script recently.

**What you expected to happen**:

`./hack/.ci/prepare_release` should not fail.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/hack/.ci/prepare_release
+++ b/hack/.ci/prepare_release
@@ -50,6 +50,10 @@ cd /usr/local/go/src
 echo "Executing make on go $GOLANG_VERSION"
 ./make.bash > /dev/null 2>&1
 
+# Make the newly "installed from source" go the default one
+export PATH="/usr/local/go/bin:$PATH"
+export GOROOT="/usr/local/go"
+
 export GOPATH="$(mktemp -d)"
 export GOBIN="$GOPATH/bin"
 export PATH="$GOBIN:$PATH"
@@ -73,4 +77,3 @@ if ! make generate; then
 fi
 cd "$cur_dir"
 cp -RT "$REPO_PATH/" "$repo_root_dir/"
-

